app/questionnaire/router.py

```diff
@@ -258,6 +268,121 @@ def _get_return_to_location_url(
                 "questionnaire.submit_questionnaire", _anchor=return_to_answer_id
             )
 
+    def _get_return_to_for_grand_calculated_summary(
+        self,
+        *,
+        return_to: str | None,
+        return_to_block_id: str | None,
+        location: Location,
+        routing_path: RoutingPath,
+        is_for_previous: bool,
+        return_to_answer_id: str | None = None,
+    ) -> str | None:
+        """
+        Builds the return url for a grand calculated summary,
+        and accounts for it possibly being in a different section to the calculated summaries it references
+        """
+        grand_calculated_summary_section = location.section_id
+        # if return_to_block_id is invalid and no block is found, the section doesn't update but can_access_location will be false, so it still returns None
+        if return_to_block_id and self._schema.get_block(return_to_block_id):
```

Comment:
For clarity use `self._schema.is_block_valid`

Reply:
updated 👍